hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/kafka/KafkaSqlConnector.java

```diff
@@ -201,6 +201,7 @@ private Vertex writeProcessor(@Nonnull DagBuildContext context) {
                         :
                         KafkaProcessors.<Entry<Object, Object>, Object, Object>writeKafkaP(
                                 new DataConnectionRef(table.dataConnectionName()),
+                                table.kafkaProducerProperties(),
```

Comment:
`kafkaProducerProperties` is never null, will this always fail in https://github.com/hazelcast/hazelcast/pull/24333/files#diff-cf5835a2445b6aac7f4eb891ef0284241c68fc2c7331c0b2715c07f4f3695bd6R186?

Reply:
Sorry, didn't understand -- why shouldn't it fail? There is another condition to fail

Reply:
what happens when you create SHARED kafka data connection and MAPPING with empty options (the only allowed way currently for shared connection)?

Reply:
I will answer myself: you will be able to read from such mapping but not write to it. This is contradictory to PR description:
> A shared Kafka connection should not accept any additional properties

Actually, you allow to read also from shared data connection with defined mapping-level properties. Update either the description or the implementation.

>Only method with properties used should fail.
You check only if the connection is shared and not if the properties are empty.